src/config/from_document.rs

```diff
@@ -237,9 +239,13 @@ fn to_enum(enum_type: EnumType) -> config::Type {
         .collect();
     config::Type { variants: Some(variants), ..Default::default() }
 }
-fn to_input_object(input_object_type: InputObjectType) -> Valid<config::Type, String> {
+fn to_input_object(
+    input_object_type: InputObjectType,
+    directives: &[Positioned<ConstDirective>],
+) -> Valid<config::Type, String> {
     to_input_object_fields(&input_object_type.fields)
-        .map(|fields| config::Type { fields, ..Default::default() })
+        .fuse(Protected::from_directives(directives.iter()))
+        .map(|(fields, protected)| config::Type { fields, protected, ..Default::default() })
```

Comment:
Needed otherwise we would not have access to `protected` directive on `update_protected` for `input` types.